lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -731,6 +731,8 @@ int ReadDxcOpts(const OptTable *optionTable, unsigned flagsToInclude,
   else
     opts.OptLevel = 3;
   opts.OptDump = Args.hasFlag(OPT_Odump, OPT_INVALID, false);
+  opts.FiniteMathOnly =
+      Args.hasFlag(OPT_ffinite_math_only, OPT_fno_finite_math_only, false);
```

Comment:
Can we instead set this value right into the SpirvOptions so that we don't need a separate bool in the HLSLOptions?

Reply:
The idea was have this option higher-up, as it could impact both SPIRV and DXIL codegen, in case you'd also want to bring some optimizations depending on this flag. (like the general optimization level flag which impacts both backends)
Let me know if you prefer for this option to be spirv only.

Reply:
We already handle this option for DXIL. All flags marked `CoreOption` are identical matches to clang flags and we handle them by passing the flag into Clang's option parsing (search for `CopyArgsToWStrings`).

We don't need any special handling for the option for DXIL.

An alternate approach SPIR-V could take would be to read the flag values out of the Clang Language Options. The parsing code today does store the value into LangOpts so it is probably already available without additional handling.

Reply:
Ahh I see, completely missed that! Thanks for the explanation!